client/src/components/Input/EndpointMenu/PromptLibrary.tsx

```diff
@@ -0,0 +1,213 @@
+import { BookOpen, SearchIcon } from 'lucide-react';
+import axios from 'axios';
+
+interface Prompt {
+  id: number;
+  open: boolean;
+  presetId?: null;
+  title: string;
+  chatGptLabel: string;
+  endpoint: string; //"openAI"
+  model: string; //"gpt-4-0613"
+  frequency_penalty: number;
+  jailbreak: boolean;
+  tags: string[];
+  presence_penalty: number;
+  promptPrefix: string;
+  temperature: number;
+  top_p: number;
+}
+
+function NavBar({ items, currentPage, handlePageChange }) {
+  const itemsPerPage = 10;
+  const maxPage = Math.ceil(items.length / itemsPerPage);
+
+  return (
+    <nav
+      className="mt-2 flex items-center justify-between bg-white px-4 py-3 dark:bg-gray-800 sm:px-6 md:rounded-md"
+      aria-label="Pagination"
+    >
+      <div className="hidden sm:block">
+        <p className="text-sm text-gray-700 dark:text-white">
+          Showing <span className="font-medium">{(currentPage - 1) * itemsPerPage + 1}</span> to{' '}
+          <span className="font-medium">{Math.min(currentPage * itemsPerPage, items.length)}</span>{' '}
+          of <span className="font-medium">{items.length}</span> results
+        </p>
+      </div>
+      <div className="flex flex-1 justify-between sm:justify-end">
+        <a
+          onClick={() => (currentPage !== 1 ? handlePageChange(currentPage - 1) : null)}
+          className={`relative inline-flex items-center rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus-visible:outline-offset-0 ${
+            currentPage === 1 ? 'cursor-not-allowed opacity-50' : ''
+          }`}
+        >
+          Previous
+        </a>
+        <a
+          onClick={() => (currentPage !== maxPage ? handlePageChange(currentPage + 1) : null)}
+          className={`relative ml-3 inline-flex items-center rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus-visible:outline-offset-0 ${
+            currentPage === maxPage ? 'cursor-not-allowed opacity-50' : ''
+          }`}
+        >
+          Next
+        </a>
+      </div>
+    </nav>
+  );
+}
+
+export function PromptLibrary({ onSelect }) {
+  const prompts: Prompt[] = [];
+  const [promptLibrary, setPromptLibrary] = useState(prompts);
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 8;
+  const [query, setQuery] = useState('');
+
+  // TODO: Replace with meilisearch
+  const filteredPrompts = promptLibrary.filter((prompt) => {
+    const search = query.toLowerCase();
+    return (
+      // Search Title
+      prompt.title.toLowerCase().includes(search) ||
+      // Search Prompt
+      prompt.promptPrefix.toLowerCase().includes(search) ||
+      // Search Tags
+      prompt.tags.some((tag) => tag.toLowerCase().includes(search))
+    );
+  });
+
+  // Todo: move to state management
+  useEffect(() => {
+    axios({
+      method: 'get',
+      url: '/api/promptLibrary',
```

Comment:
Why are we making an api call directly from the UI? Please use the data provider with react query. I wrote [an article for this](https://www.danorlandoblog.com/building-data-services-for-librechat-with-react-query/) to help those who are unfamiliar with the data services pattern.

Reply:
Its also worth noting that the data-provider package allows for the apis to be used from the admin. I have ideas for an admin page where admins can create and edit prompts for this prompt library.